packages/laravel/src/Concerns/HasSharedProperties.php

```diff
@@ -25,6 +25,18 @@ public function share(string|array|Arrayable $key, mixed $value = null): static
         return $this;
     }
 
+    /**
+     * Shares data to every response if the condition is true.
+     */
+    public function shareWhen(bool|\Closure $condition, string|array|Arrayable $key, mixed $value = null): static
+    {
+        if ($condition) {
```

Comment:
If `$condition` is given as a closure here, it will always pass even if the closure is supposed to return `false`.

Reply:
Oops, fixed it.